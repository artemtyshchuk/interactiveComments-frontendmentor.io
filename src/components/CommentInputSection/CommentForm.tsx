import { UserImage } from "components/UserImage";
import styles from "./CommentInputSection.module.scss";
import { TextArea } from "components/TextArea";
import { MainButton } from "components/MainButton";

interface CommentFormProps {
  onSubmit: (event: React.FormEvent) => void;
  onChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
  userImage: string;
  value: string;
  loading: boolean;
  content: string;
  error: string | null;
}

export const CommentForm = ({
  onSubmit,
  onChange,
  userImage,
  value,
  loading,
  content,
  error,
}: CommentFormProps) => {
  return (
    <form className={styles.commentInput} onSubmit={onSubmit}>
      <div className={styles.userImageContainer}>
        <UserImage image={userImage} alt="userImage" inputImage />
      </div>
      <div className={styles.inputContainer}>
        <TextArea
          placeholder="Add a comment..."
          value={value}
          onChange={onChange}
        />
      </div>
      <div className={styles.sendButtonContainer}>
        <MainButton
          mainButtonText={loading ? "Sending..." : "Send"}
          content={content}
          type="submit"
        />
      </div>
      {error && <p className={styles.error}>{error}</p>}
    </form>
  );
};
