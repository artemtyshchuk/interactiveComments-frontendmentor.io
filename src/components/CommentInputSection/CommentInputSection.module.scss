.commentInput {
  display: grid;
  grid-template-columns: auto 1fr auto;
  column-gap: 16px;
  padding: 24px;
  align-items: start;
  background-color: #fff;
  width: 100%;
  border-radius: 8px 8px 0 0;
  box-sizing: border-box;
}

@media (max-width: 768px) {
  //577
  .commentInput {
    display: grid;
    grid-template-columns: auto 1fr auto;
    column-gap: 16px;
    padding: 18px;
    align-items: start;
    background-color: #fff;
    width: 100%;
    border-radius: 8px 8px 0 0;
    box-sizing: border-box;
  }
}

@media (max-width: 576px) {
  .commentInput {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 1fr auto;
    row-gap: 16px;
    padding: 18px;
    background-color: #fff;
    width: 100%;
    border-radius: 8px 8px 0 0;
    box-sizing: border-box;
    align-items: center;
  }

  .userImageContainer {
    grid-row: 2/3;
    grid-column: 1/2;
  }

  .inputContainer {
    grid-row: 1/2;
    grid-column: 1/3;
  }

  .sendButtonContainer {
    justify-self: end;
    grid-row: 2/3;
  }
}
