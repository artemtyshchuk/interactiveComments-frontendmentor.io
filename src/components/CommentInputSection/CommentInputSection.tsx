import React, { useState } from "react";
import { CommentsType, UserType } from "types";
import { usePostComment } from "hooks/usePostComment";
import { CommentForm } from "./CommentForm";

interface CommentInputSectionProps extends UserType {
  onSubmit: (newComment: CommentsType) => void;
}

export const CommentInputSection = ({
  image,
  onSubmit,
  username,
}: CommentInputSectionProps) => {
  const [comment, setComment] = useState<string>("");
  const { postComment, loading, error } = usePostComment();

  const handleSubmit = async (event: React.FormEvent) => {
    event.preventDefault();
    await postComment(comment, image, username, onSubmit);
    setComment("");
  };

  return (
    <>
      <CommentForm
        onSubmit={handleSubmit}
        onChange={(event: React.ChangeEvent<HTMLTextAreaElement>) =>
          setComment(event.target.value)
        }
        userImage={image.png}
        value={comment}
        loading={loading}
        content={comment}
        error={error}
      />
    </>
  );
};
