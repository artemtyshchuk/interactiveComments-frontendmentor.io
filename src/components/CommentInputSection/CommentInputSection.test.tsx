import { render, screen } from "@testing-library/react";
import { CommentInputSection } from "./CommentInputSection";

describe("CommentInputSection", () => {
  it("should render correctly", () => {
    render(
      <CommentInputSection
        onSubmit={() => {}}
        username={"username"}
        image={{ png: "", webp: "" }}
      />
    );

    expect(screen.getByRole("img")).toMatchSnapshot();
  });
});
