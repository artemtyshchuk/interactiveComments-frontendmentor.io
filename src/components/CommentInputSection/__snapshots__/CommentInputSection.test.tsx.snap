// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentInputSection should render correctly 1`] = `
<img
  alt="userImage"
  class="userImage"
  src=""
/>
`;
